src/lua/protobuf_wireformat.lua

```diff
@@ -97,6 +141,29 @@ local function encode_varint(value)
     return ffi.string(buf, size)
 end
 
+-- Decode input value as VARINT without a tag.
+--
+-- Input value type: string
+--
+-- This is a helper function to decode tag and data values.
+local function decode_varint(ctx)
+    local msb
+    local res = 0
+    local len = 0
+    repeat
+        local payload = ffi.cast('uint64_t', string.byte(ctx.val, ctx.pos))
+        if len >= 10 or payload == nil then
```

Comment:
I would define 10 as a constant and clarify it somehow.

Reply:
Added a constant `MAX_VARINT_LEN` and added a comment to it.